parser.py

```diff
@@ -54,31 +54,32 @@ def tweet_json_to_markdown(tweet, username, archive_media_folder, output_media_f
             if 'url' in url and 'expanded_url' in url:
                 body = body.replace(url['url'], url['expanded_url'])
     # replace image URLs with markdown image links to local files
-    if 'entities' in tweet and 'media' in tweet['entities']:
-        for media in tweet['entities']['media']:
+    if 'extended_entities' in tweet and 'media' in tweet['extended_entities']:
+        markdown = ''
+        for media in tweet['extended_entities']['media']:
             if 'url' in media and 'media_url' in media:
                 original_url = media['url']
                 original_expanded_url = media['media_url']
                 original_filename = os.path.split(original_expanded_url)[1]
                 local_filename = os.path.join(archive_media_folder, tweet_id_str + '-' + original_filename)
                 new_url = output_media_folder_name + tweet_id_str + '-' + original_filename
+                markdown += '' if not markdown and body.startswith(original_url) else '\n\n'

```

Comment:
Why would body startwith original_url? I don't understand this bit.

Reply:
body starts with original_url if that's all that it contains. This could probably be == rather than .startswith() - I'll try that out and see if it changes any output..

Reply:
Oh I see. This is inserting newlines after the tweet text if there is any, otherwise not. Thanks.

Reply:
I hadn't understood how Twitter uses a single t.co link to represent multiple images. Got it.